### PR TITLE
fix(chat): add tooltips in publish and unpublish (Issue #1421, #1447, #1450, #1478)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

Need to add tooltip for folders names inside (un)publish request for prompt/chat folders
currently tooltip pointing to the center of the textbox
Note:Different font sizes for tooltips for Publish to and prompt/conversation name on the publication request details
No tooltip displayed on hover chat/prompt/folder name on unpublish pop-up
Tooltip need to fit the request details area, not go beyond borders to prompt/chat side pannels
Path should be end with 3 dots if it is not fit the section length, tooltip should be shown on hover

Issues:

- Issue #1421 ,
- Issue #1447 ,
- Issue #1450 ,
- Issue #1478 

**UI changes**

Снимок экрана 2024-06-09 в 23 23 17
Снимок экрана 2024-06-09 в 23 23 29
Снимок экрана 2024-06-09 в 23 23 53
Снимок экрана 2024-06-09 в 23 24 18
Снимок экрана 2024-06-09 в 23 24 31

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
No release type found in pull request title " fix(chat): add tooltips in publish and unpublish (Issue #1421, Issue #1447, Issue #1450)". Add a prefix to indicate what kind of release this pull request corresponds to. For reference, see https://www.conventionalcommits.org/

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/483732)

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/483735)

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/PublicationResources.tsx [hunk 1]
```diff
@@ -38,13 +38,15 @@ interface PublicationResources {
   resources: PublicationResource[];
   forViewOnly?: boolean;
   rootFolder?: ShareEntity;
+  isHaveTooltip?: boolean;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/PublicationResources.tsx]: It's more correct `hasTooltip`, but better `showTooltip`
```suggestion
  showTooltip?: boolean;
```

File: apps/chat/src/components/Folder/Folder.tsx [hunk 2]
```diff
@@ -117,6 +117,7 @@ export interface FolderProps<T, P = unknown> {
   noCaretIcon?: boolean;
   itemComponentClassNames?: string;
   canAttachFolders?: boolean;
+  isHaveTooltip?: boolean;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Folder/Folder.tsx]: It's more correct `hasTooltip`, but better `showTooltip`
```suggestion
  showTooltip?: boolean;
```

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Folder/Folder.tsx [hunk 3]
```diff
@@ -915,7 +917,8 @@ const Folder = <T extends ConversationInfo | PromptInfo | DialFile>({
             >
               <Tooltip
                 tooltip={
-                  featureType === 'file' && !isNameOrPathInvalid
+                  (featureType === 'file' || showTooltip) &&
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Folder/Folder.tsx]:
```suggestion
                  (featureType === FeatureType.File || showTooltip) &&
```

Comment (Derikyan) [reply apps/chat/src/components/Folder/Folder.tsx]: corrected

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/HandlePublication.tsx [hunk 4]
```diff
@@ -181,12 +181,14 @@ export function HandlePublication({ publication }: Props) {
...
       dataQa: 'conversations-to-approve',
       Component: ConversationPublicationResources,
+      showTooltip: true,
     },
     {
       featureType: FeatureType.Prompt,
       sectionName: t('Prompts'),
       dataQa: 'prompts-to-approve',
       Component: PromptPublicationResources,
+      showTooltip: true,
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/Publish/HandlePublication.tsx:191]: Why we don't want to show tooltip for file resources?

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Chat/Publish/HandlePublication.tsx (hunk 4 above)
Comment (Derikyan) [reply apps/chat/src/components/Chat/Publish/HandlePublication.tsx:191]: refactored

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/485378)
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1540.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**